File: 575-distribute-candies/distribute-candies.java
class Solution {
    public int distributeCandies(int[] candyType) {
        Arrays.sort(candyType);
        int n = candyType.length;
        Set<Integer> st = new HashSet<>();;
        for (int candy : candyType) {
            st.add(candy);
        }

        int m = st.size();

        return Math.min(m,n/2);
    }
}

File: 575-distribute-candies/distribute-candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        set<int> st (candyType.begin() , candyType.end());
        int n = candyType.size();
        int m = st.size();

        return min(m , n/2);
        
    }
};